src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/VisualRuleForm.tsx

```diff
@@ -0,0 +1,238 @@
+    });
+    const commonFormStyles = getPanelFormStyles(theme, rootHeight);
+    const typeOptions: Array<IDropdownOption> = useMemo(
+        () =>
+            Object.keys(DTDLPropertyIconographyMap).map((mappingKey) => ({
+                key: `value-type-${DTDLPropertyIconographyMap[mappingKey].text}`,
+                text: DTDLPropertyIconographyMap[mappingKey].text,
+                data: { icon: DTDLPropertyIconographyMap[mappingKey].icon }
+            })),
+        []
+    );
+    // TODO: Wire this up to actual form data
+    const formData = {
+        propertyType: 'boolean'
+    };
+
+    // Contexts
+    const {
+        config,
+        sceneId,
+        adapter,
+        setVisualRuleFormMode,
+        state: { selectedElements, selectedBehavior }
+    } = useContext(SceneBuilderContext);
+
+    // TODO: Update form callbacks to wire them to form state
+    const onTabValidityChange = useCallback(
+        (_tabName: TabNames, _state: IValidityState) => {
+            return;
+        },
+        []
+    );
+
+    const onPropertyChange = useCallback(
+        (_newPropertyExpression: PropertyExpression) => {
+            return;
+        },
+        []
+    );
+
+    const onTypeChange = useCallback(
+        (
+            _event: React.FormEvent<HTMLDivElement>,
+            _option?: IDropdownOption<any>,
+            _index?: number
+        ) => {
+            return;
+        },
+        []
+    );
+
+    const customStyles = mergeStyleSets({
+        description: {
+            fontSize: FontSizes.size14,
+            color: theme.palette.neutralSecondary
+        }
+    });
+
+    const onInternalModeChanged = useCallback((internalMode) => {
+        if (internalMode === 'INTELLISENSE') {
+            setPropertyTypeDropdownEnabled(true);
+        } else {
+            setPropertyTypeDropdownEnabled(false);
+        }
+    }, []);
+
+    return (
+        <>
+            <div className={commonFormStyles.content}>
+                <div className={commonFormStyles.header}>
+                    <Stack tokens={{ childrenGap: 12 }}>
+                        <div className={customStyles.description}>
+                            {t('3dSceneBuilder.visualRuleForm.formDescription')}
+                        </div>
+                        <TextField
+                            label={t('displayName')}
+                            // value={FormState}
+                            required={true}
+                            onChange={(_e, newValue) => {
+                                // Alerts change to visual rules
+                                onTabValidityChange('Alerts', {
+                                    isValid: !!newValue
+                                });
+                                // Add dispatch for state here
+                            }}
+                            styles={classNames.subComponentStyles.textField}
+                        />
+                        <ModelledPropertyBuilder
```

Comment:
probably it is not related to the scope of this PR but I am not able to list the twin properties in this component for some reason. Did you experience it also @pasanchMSFT ?
![image](https://user-images.githubusercontent.com/45217314/194157262-dc37a37d-0d68-4d57-bf6f-5182d5e98d14.png)

Reply:
I will take a look at this when I have real data.